src/domain/event/eventInfo/EventInfo.tsx

```diff
@@ -123,7 +156,11 @@ const EventInfo: React.FC<Props> = ({ event }) => {
         </InfoWithIcon>
 
         {/* Other event times */}
-        <OtherEventTimes event={event} />
+        {eventType === EventType.EVENT ? (
+          <OtherEventTimesContainer event={event} />
```

Comment:
eventType could be passed as parameter, so we should not need separate `OtherEventTimesContainer` and `OtherCourseTimesContainer` components?

Reply:
No sure how that would work

Reply:
These containerr components are where because of the different hooks

Reply:
hook is just a function, you can use different function depending on the event param?
`const {field1, field2 } = param ? useHook1(event) : useHook2(event); `

Reply:
hooks cannot be used conditionally

Reply:
https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Reply:
Ok forgot about that but one option is to put the condition inside custom hook as I think that hook can call another hook conditionally without react complaining

Reply:
And how do you make sure that it is not gonna break now or in the future?

Reply:
and putting the condition inside another custom hooks is not gonna change anything, react still complains

Reply:
Could there be a chance to use skip-value? https://stackoverflow.com/questions/59489097/use-query-hook-react-apollo-conditional-call
Not sure how to do this, genreally there should be a way to avoid copy-pasting (now components are 95% same)
Maybe it can be done much easier when courses and events are unified on backend. And this can be resolved & merged

Reply:
They could be moved to same file to avoid some repetition with imports but basically the repetition is one line (using component with different props).

maybe skip value could be used but I think it would make the component overly complicated (which props are passed from hooks to the component?), and if more EventTypes would be added itwould be event more complicated.

And of course I would like to wait for the new linked-events API. Maybe then we can come up with better implementation with fetching events and courses.

And we are basically creating helper-components that use different hooks and pass different props to the other component, I would harly call that "copy-paste".

Something to read:
https://kentcdodds.com/blog/aha-programming
https://medium.com/@thisdotmedia/the-cost-of-premature-abstraction-b5d71ffd6400